pkg/controller/authentication/deployment.go

```diff
@@ -567,11 +576,21 @@ func generateManagerDeploymentObject(instance *operatorv1alpha1.Authentication,
 	ldapCACert := instance.Spec.AuthService.LdapsCACert
 	routerCertSecret := instance.Spec.AuthService.RouterCertSecret
 
+	metaLabels := pkgCommon.MergeMap(map[string]string{"app": deployment}, instance.Spec.Labels)
+	podMetadataLabels := map[string]string{
+		"app":                        deployment,
+		"k8s-app":                    deployment,
+		"component":                  deployment,
+		"app.kubernetes.io/instance": "platform-identity-management",
+		"intent":                     "projected",
+	}
+	podLabels := pkgCommon.MergeMap(podMetadataLabels, instance.Spec.Labels)
```

Comment:
instead of swapping the arguments in this function, In the MergeMap function body i have swapped it, meaning that i have considered mergeMap as a source(instance.Spec.Labels) to add back podMetadataLables to mergeMap, so that we preserve the podMetadataLabels